extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -470,4 +480,81 @@ public void refreshPresetChannel() {
                 .sorted(Comparator.comparing(StateOption::getValue)).collect(Collectors.toList());
         stateOptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_PRESET), stateOptions);
     }
+
+    public void handleGroupUpdated(BoseSoundTouchConfiguration masterPlayerConfiguration) {
+        String deviceId = getMacAddress();
+        
+        if (masterPlayerConfiguration != null && masterPlayerConfiguration.macAddress != null) {
+            // Stereo pair
+            if (Objects.equals(masterPlayerConfiguration.macAddress, deviceId)) {
+                if (getThing().getThingTypeUID().equals(BST_10_THING_TYPE_UID)) {
+                    // change thing type to Stereo Pair
+                    logger.debug("{}: Stereo Pair was created and this is the master device. Changing ThingType to: {}",
+                            getDeviceName(), BST_GROUP_THING_TYPE_UID);
+                    changeThingType(BST_GROUP_THING_TYPE_UID, getConfig());
+                } else if (getThing().getThingTypeUID().equals(BST_GROUP_THING_TYPE_UID)) {
+                    // updated group, this is master device, thing type is correct - nothing more to do
+                } else {
+                    // unsupported thing type - LOG
+                    logger.debug("{}: Unsupported operation for player of type: {}", getDeviceName(),
+                            getThing().getThingTypeUID());
+                }
+            } else {
+                if (getThing().getThingTypeUID().equals(BST_10_THING_TYPE_UID)) {
+                    // Stereo Pair was created and this is NOT the master device
+                    logger.debug("{}: Stereo Pair was created and this is NOT the master device.", getDeviceName());
+                    updateThing(editThing().withChannels(Collections.emptyList()).build());
+                    changeThingType(BST_GROUP_THING_TYPE_UID, getConfig());

```

Comment:
Changing the ThingType will recreate all channels for the given ThingTypeUID. I would suggest to not switch thing type in this scenario. The ThingType did actually not change. Its more of the operation mode which did change.
In case you want to switch here because UI should show a different icon, why not changing the state of the corresponding item?

Reply:
Do you mean - only for the 'slave' case, or also for the master ?
Because my understanding is that the ThingType for the master actually changes (it stops being a SoundTouch 10 device and starts being a Stereo Pair device)

Reply:
But it's still the same device. If I'm not mistaken thing type migration was meant to be used when some handler knows a more specific thing type for the device. Here the same device just turns into a different operation mode. Or does it fundamentally change its capabilities in terms of available channels?